test/unit/resources/th_main.js

```diff
@@ -800,24 +793,27 @@ var TestUniverseMixins = {
       var willDownloadSnippets = !!state.inboxHasNewMessages &&
                                  testAccount.type !== 'pop3';
 
-      testAccount.expect_releasedAccountSave();
-      testAccount.eOpAccount.expect_saveAccountState_end(
-        testAccount.type !== 'pop3' ? 'checkpointSync' : 'syncComplete');
-      if (testAccount.eOpAccount.ignore_releaseConnection) {
-        testAccount.eOpAccount.ignore_releaseConnection();
-      }
+      testAccount.expect('releasedAccountSave');
+      testAccount.eOpAccount.expect('saveAccountState_end', {
+        reason: testAccount.type !== 'pop3' ? 'checkpointSync' : 'syncComplete'
+      });
+
+      // testAccount.eOpAccount.ignore_releaseConnection();
 
-      this.eCronSync.expect_syncAccountHeaders_end(testAccount.accountId);
+      this.eCronSync.expect('syncAccountHeaders_end',
+                            { accountId: testAccount.accountId });
 
       if (willDownloadSnippets) {
         // Since we're going on to download something, there will be a server
         // job and although the connection will be released, it will then be
         // immediately reused for the job.
         if (testAccount.eImapAccount) {
-          testAccount.eImapAccount.expect_releaseConnection();
+          // testAccount.eImapAccount.expect('releaseConnection');
```

Comment:
Hooray for the horrible connection-management hassles being done with!  (Note: fine to leave this commented out as-is)